mautic/mautic
# run PHPUnit tests with a random db prefix by default

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [x]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

PR is based on 4.x, once 5.x is up-to-date, I'll change the base

This PR ensures all tests are executed with a db table prefix to ensure both tests and functionality are compatible with it.
This to prevent that code covered by a test is missing awareness of a db table prefix.

This PR adds a `BeforeFirstTestHook` to the phpunit test suite, that sets a random `MAUTIC_TABLE_PREFIX` if it isn't defined yet.
All places that check the existence of `MAUTIC_TABLE_PREFIX` are removed from the tests, as those are obsolete.

If you want to use a fixed / specific prefix in the tests, it's still possible by setting the env var in the `app/phpunit.xml.dist`  file.



#### Steps to test this PR:

* nothing really to test, if tests pass, all good :)


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.
